### PR TITLE
Move lerp for the engine types into dart:sky

### PR DESCRIPTION
**Pull request by abarth**
This patch just adds them to dart:sky. A future patch will actually use them in
the framework once we've published an updated sky_engine package. Also, add
dartdoc to the classes touched in this patch.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: sky/engine/core/painting/Color.dart [hunk 1]
```diff
@@ -13,31 +26,67 @@ class Color {
...
 
+  /// Returns a new color that matches this color with the blue channel replaced with b.
   Color withBlue(int b) {
     return new Color.fromARGB(alpha, red, green, b);
   }
 
+  /// Linearly interpolate between two colors
+  ///
+  /// If either color is null, this function linearly interpolates from a
+  /// transparent instance of othe other color.
```
Comment (abarth) [inline sky/engine/core/painting/Color.dart]: Typo

File: sky/engine/bindings/scripts/templates/dart_blink.template [hunk 2]
```diff
@@ -25,3 +25,14 @@ final Tracing tracing = new Tracing();
 View view;
 
 typedef EventListener(Event event);
+
+/// Linearly interpolate between two numbers.
+num lerpNum(num a, num b, double t) {
```
Comment (Hixie) [inline sky/engine/bindings/scripts/templates/dart_blink.template]: we should probably make this lerpDouble, and not have integers used when calling this.

File: sky/engine/core/painting/Color.dart [hunk 3]
```diff
@@ -4,8 +4,21 @@
...
+}
+
+/// An immutable 32 bit color value in ARGB
 class Color {
+  /// Construct a color from the lower 32 bits of an int
+  ///
+  /// Bits 24-31 are the alpha value.
+  /// Bits 16-23 are the red value.
+  /// Bits 4-15 are the green value.
+  /// Bits 0-3 are the blue value.
```
Comment (Hixie) [inline sky/engine/core/painting/Color.dart]: i don't believe you.

File: sky/engine/core/painting/Color.dart [hunk 4]
```diff
@@ -13,31 +26,67 @@ class Color {
                 ((b & 0xff) << 0)) & 0xFFFFFFFF);
 
   final int _value;
+  /// A 32 bit value representing this color
+  ///
+  /// Bits 24-31 are the alpha value.
+  /// Bits 16-23 are the red value.
+  /// Bits 4-15 are the green value.
+  /// Bits 0-3 are the blue value.
```
Comment (Hixie) [inline sky/engine/core/painting/Color.dart]: here too.

**Comment by Hixie** [comment]
LGTM

**Pull request merged**